### DISCUSSION TITLE
Does failure of 3.1.1 imply failure of 3.1.2?

### DISCUSSION THREAD

**Discussion by stevefaulkner** [w3c/wcag#3850] [Q&A]
[3.1.1 Language of Page](https://www.w3.org/TR/WCAG21/#language-of-page) fails if the default human language of each Web page cannot be programmatically determined. Success Criterion [3.1.2 Language of Parts](https://www.w3.org/TR/WCAG21/#language-of-parts) requires that "The human language of each passage or phrase in the content can be programmatically determined" which implies that if no language is identified programmatically for 3.1.1 the 3.1.2 requirement cannot be satisfied unless the human language of each passage or phrase is identified using a `lang` attribute.
/cc @patrickhlauke

**Comment by JAWS-test** [comment]
According to the wording of the SC, this would be the case, but the first sentence of the Understanding reads 

> The intent of this Success Criterion is to ensure that user agents can correctly present content written in **multiple languages**

As long as different languages are not used in one page at the same time, it should only be a violation of 3.1.1.

**Comment by frex65** [comment]
It seems to me that if 3.1.2 can be said to fail if a `lang` attribute hasn't been set on either the `html` element or on **each and every** element containing a passage or phrase of text, whether or not it constitutes a change from the default language, then this would effectively render 3.1.1 redundant.
If the `html` element does not contain a `lang` attribute, is it safe to assume that the document has no default language, and therefore that none of the passages and phrases in the document are written in a default language? Even if that's the case, this seems to render 3.1.1 redundant.

**Comment by stevefaulkner** [comment]
> According to the wording of the SC, this would be the case, but the first sentence of the Understanding reads

Unless I am mistaken, this issue is that the SC is **normative** the understanding is _informative_.

**Comment by patrickhlauke** [comment]
I’d say yes, and have said as much on w3c mailing list in the past (can’t search for reference right now). Even if you had no `lang` on html element but on body, it would leave the `<title>` as indeterminate language.

**Comment by alastc** [comment]
I vaguely remember the a discussion about why 3.1.1 was level A when the issues raised by 3.1.2 were a higher priority from a user's point of view, and the upshot was (roughly): 3.1.1 is level A because you need the default set in order for element-level `lang`s to actually differentiate the language.

IMO, by the SC text, you could infer that any fail of 3.1.1 means 3.1.2 will fail. 

Personally, I see 3.1.2 as a 'change of language' requirement, and in my company's internal practice we only fail 3.1.2 if there's is a change of language, and will report that regardless of 3.1.1. If they already fail the level A SC, that's already perceived as a higher priority.

**Comment by patrickhlauke** [comment]
> Personally, I see 3.1.2 as a 'change of language' requirement, and in my company's internal practice we only fail 3.1.2 if there's is a change of language, and will report that regardless of 3.1.1. If they already fail the level A SC, that's already perceived as a higher priority.

This is one of those "cascades of fail" situations. If you want to be overly specific, then it would make sense to also fail under 3.1.2, because technically it does fail. Practically though, the error lies in 3.1.1, and if that's already reported as a failure, then there may be a sense of not needing to "twist the knife" further. I personally fall into the former camp of reporting everything that's a fail, even if it's because of another failure and that's been reported as fail already.

**Comment by fstrr** [comment]
This issue is labelled as a discussion, so we’re moving this to Discussions. There doesn’t seem to be an update to make to the documentation, but if that changes, we can move it back to the issues list.
